components/atoms/button/components/button/button.js

```diff
@@ -84,7 +86,7 @@ const StyledButton = styled.button`
   vertical-align: middle;
   white-space: nowrap;
   margin: ${({ centre }) => (centre ? 'auto' : 0)};
-  padding: ${({ theme }) => theme.SPACING(3, 6)};
+  padding: ${({ theme, noPadding }) => (noPadding ? '0' : theme.SPACING(3, 6))};
```

Comment:
Shouldn't this `noPadding` also work for other sizes? Look at line 103 in the file which we're defining padding for `XS` size.